lib/transfer.c

```diff
@@ -1779,6 +1779,12 @@ CURLcode Curl_retry_request(struct connectdata *conn,
     retry = TRUE;
   }
   if(retry) {
+#define CONN_MAX_RETRIES 5
+    if(conn->retrycount++ >= CONN_MAX_RETRIES) {
+      failf(data, "Connection died, tried %d times before giving up",
+            CONN_MAX_RETRIES);
+      return CURLE_SEND_ERROR;
```

Comment:
Is there any reason why you chose CURLE_SEND_ERROR instead of CURLE_COULDNT_CONNECT

Reply:
It manages to connect, it just can't *use* the connection. I couldn't really make up my mind which error code that would be....